### PR TITLE
Use boto3 instead of boto2

### PR DESCRIPTION
**Pull request by marek-obuchowicz**
As boto3 is handling "throttling" issue automatically, using new library version allows zmon-agent to work correctly with AWS accounts consisting of more elements.

### CODE REVIEW

**Comment by hjacobs** [comment]
@marek-obuchowicz thanks for the PR, could you check the flake8 formatting please?

**Comment by marek-obuchowicz** [comment]
Please note: @Jan-M should check RDS functionality before merging it, as we don't use it on our account

File: zmon-agent.py [hunk 1]
```diff
@@ -261,7 +266,12 @@ def main():
 
     if not args.region:
         logging.info("Trying to figure out region...")
-        region = boto.utils.get_instance_metadata()['placement']['availability-zone'][:-1]
+        try:
+            response = requests.get('http://169.254.169.254/latest/meta-data/placement/availability-zone', timeout=2)
```
Comment (hjacobs) [inline zmon-agent.py]: this looks wrong, you still need to cut off the last char to get the region (now you have the AZ)

File: zmon-agent.py [hunk 2]
```diff
@@ -261,7 +266,12 @@ def main():
...
+        except:
+            logging.error("Region was not specified as a parameter and can not be fetched from instance meta-data!")
+            raise
+        region = response.text
```
Comment (hjacobs) [inline zmon-agent.py]: response.text[:-1]

File: zmon-agent.py [hunk 3]
```diff
@@ -50,39 +43,60 @@ def get_hash(ip):
...
-            if str(i._state) != 'running(16)':
+            if str(i['State']['Name']) != 'running':
                 continue
 
             try:
-                user_data = base64.b64decode(str(i.get_attribute('userData')["userData"]))
+                user_data_response = aws_client.describe_instance_attribute(InstanceId=i['InstanceId'],
+                                                                            Attribute='userData')
+                user_data = base64.b64decode(user_data_response['UserData']['Value'])
                 user_data = yaml.load(user_data)
```
Comment (hjacobs) [inline zmon-agent.py]: we should use yaml.safe_load here

**Pull request merged**